### PR TITLE
Fix for memory leak issue #42

### PR DESCRIPTION
**Pull request by midmohan**

### CODE REVIEW

**Comment by sdroege** [comment]
See my comment here: https://gitlab.freedesktop.org/gstreamer/gstreamer-sharp/merge_requests/16#note_412876

**Comment by sdroege** [comment]
Also https://github.com/GLibSharp/GtkSharp/pull/41 might be interesting for you too, without that you'd also get basically every element leaked.

https://github.com/GtkSharp/GtkSharp/ has a different "solution" to this (related to that other issue/PR you linked to over there), which is wrong from what I can see.

File: Source/glib/Bytes.cs [hunk 1]
```diff
@@ -158,6 +158,7 @@ public FinalizerInfo (IntPtr handle)
 			public bool Handler ()
 			{
 				g_bytes_unref (handle);
+                GLib.Timeout.Remove(timeoutHandlerId);
```
[the remark this answers is not in the document]
Comment (thiblahute) [reply Source/glib/Bytes.cs]: Also, you forgot to add the `timeoutHandlerID` field in the `FinalizerInfo` for Bytes.

**Comment by midmohan** [comment]
> Also please update the generated code now that it is checked out in tree please :-)

Are there more generated files that has the finalizerInfo class, my search dint returned any. Can you please let me know , which files you are referring to.

**Comment by thiblahute** [comment]
> Are there more generated files that has the finalizerInfo class, my search dint returned any. Can you please let me know , which files you are referring to.

All the files in the various `generated/` folder, I made substantial changes today and now generated files are being checked out in the source tree, the same way we do in in `gstreamer-sharp`.

**Comment by thiblahute** [comment]
As discussed in the gstreamer-sharp counter part of that PR, I think we should rather fix the main issue in the way `GSource` is wrapped and implement what is being described [here](https://gitlab.freedesktop.org/gstreamer/gstreamer-sharp/merge_requests/16#note_412913)

**Comment by midmohan** [comment]
> As discussed in the gstreamer-sharp counter part of that PR, I think we should rather fix the main issue in the way `GSource` is wrapped and implement what is being described [here](https://gitlab.freedesktop.org/gstreamer/gstreamer-sharp/merge_requests/16#note_412907)
> 
> Do you think you could work on that?

Let me take a stab at that.

**Comment by midmohan** [comment]
raised https://github.com/GLibSharp/GtkSharp/pull/45

**Comment by thiblahute** [comment]
Closing in favor of https://github.com/GLibSharp/GtkSharp/pull/45

**Pull request closed without merging**